src/Uno.UI.Runtime.Skia.Wpf/Input/HostPointerHandler.cs

```diff
@@ -0,0 +1,254 @@
+			}
+			catch (Exception e)
+			{
+				this.Log().Error("Failed to raise PointerPressed", e);
+			}
+		}
+
+		private void HostOnMouseUp(object sender, MouseButtonEventArgs args)
+		{
+			try
+			{
+				_host.XamlRoot?.VisualTree.ContentRoot.InputManager.RaisePointerReleased(_previous = BuildPointerArgs(args));
+			}
+			catch (Exception e)
+			{
+				this.Log().Error("Failed to raise PointerReleased", e);
+			}
+		}
+
+		private IntPtr OnWmMessage(IntPtr hwnd, int msg, IntPtr wparamOriginal, IntPtr lparamOriginal, ref bool handled)
+		{
+			var wparam = (int)(((long)wparamOriginal) & 0xFFFFFFFF);
+			var lparam = (int)(((long)lparamOriginal) & 0xFFFFFFFF);
+
+			static short GetLoWord(int i) => (short)(i & 0xFFFF);
+			static short GetHiWord(int i) => (short)(i >> 16);
+
+			switch (msg)
+			{
+				case Win32Messages.WM_DPICHANGED:
+					break;
+				case Win32Messages.WM_MOUSEHWHEEL:
+				case Win32Messages.WM_MOUSEWHEEL:
+					{
+						var keys = (MouseModifierKeys)GetLoWord(wparam);
+
+						// Vertical: https://docs.microsoft.com/en-us/windows/win32/inputdev/wm-mousewheel
+						// Horizontal: https://docs.microsoft.com/en-us/windows/win32/inputdev/wm-mousehwheel
+
+						var l = (int)lparam;
+						var screenPosition = new Point(GetLoWord(l), GetHiWord(l));
+						var wpfPosition = _hostControl.PointFromScreen(screenPosition);
+						var position = new Windows.Foundation.Point(wpfPosition.X, wpfPosition.Y);
+
+						var properties = new PointerPointProperties
+						{
+							IsLeftButtonPressed = keys.HasFlag(MouseModifierKeys.MK_LBUTTON),
+							IsMiddleButtonPressed = keys.HasFlag(MouseModifierKeys.MK_MBUTTON),
+							IsRightButtonPressed = keys.HasFlag(MouseModifierKeys.MK_RBUTTON),
+							IsXButton1Pressed = keys.HasFlag(MouseModifierKeys.MK_XBUTTON1),
+							IsXButton2Pressed = keys.HasFlag(MouseModifierKeys.MK_XBUTTON2),
+							IsHorizontalMouseWheel = msg == Win32Messages.WM_MOUSEHWHEEL,
+							IsPrimary = true,
+							IsInRange = true,
+							MouseWheelDelta = -((int)wparam >> 16) / 40
+						}.SetUpdateKindFromPrevious(_previous?.CurrentPoint.Properties);
+
+						var modifiers = VirtualKeyModifiers.None;
+						if (keys.HasFlag(MouseModifierKeys.MK_SHIFT))
+						{
+							modifiers |= VirtualKeyModifiers.Shift;
+						}
+						if (keys.HasFlag(MouseModifierKeys.MK_CONTROL))
+						{
+							modifiers |= VirtualKeyModifiers.Control;
+						}
+
+						var point = new Windows.UI.Input.PointerPoint(
+							frameId: FrameIdProvider.GetNextFrameId(),
+							timestamp: (ulong)Environment.TickCount,
+							device: PointerDevice.For(PointerDeviceType.Mouse),
+							pointerId: 1,
+							rawPosition: position,
+							position: position,
+							isInContact: properties.HasPressedButton,
+							properties: properties
+						);
+						var ptArgs = new PointerEventArgs(point, modifiers);
+
+						_host.XamlRoot?.VisualTree.ContentRoot.InputManager.RaisePointerWheelChanged(_previous = ptArgs);
```

Comment:
Move to a separate method?

Reply:
Added as TODO, this is copied from previous code in verbatim